src/fdosecrets/widgets/AccessControlDialog.h

```diff
@@ -0,0 +1,117 @@
+/*
+ *  Copyright (C) 2013 Francois Ferrand
+ *  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+ *  Copyright (C) 2020 Aetf <aetf@unlimitedcodeworks.xyz>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSXC_FDOSECRETS_ACCESSCONTROLDIALOG_H
+#define KEEPASSXC_FDOSECRETS_ACCESSCONTROLDIALOG_H
+
+#include <QAbstractTableModel>
+#include <QDialog>
+#include <QPointer>
+#include <QPushButton>
+#include <QScopedPointer>
+#include <QSet>
+
+class Entry;
+
+namespace Ui
+{
+    class AccessControlDialog;
+}
+
+class AccessControlDialog : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit AccessControlDialog(QWindow* parent, const QList<Entry*>& items, const QString& name, uint pid);
+    ~AccessControlDialog() override;
+
+    enum class Decision
+    {
+        Undecided,
+        Allowed,
+        AllowedOnce,
+        Denied,
+        DeniedOnce,
+    };
```

Comment:
These definitions are only used in the dialog

Reply:
I wanted to make the dialog as standalone as possible, as we're going to unify/reuse it for the browser.

I'm adding the definition to `core/Global.h`. We can always split it into dedicated files if it grows beyond simple enums.